### PR TITLE
(#248) CMS Promo Image Verification tests 

### LINKED ISSUE #248: CMS: Promo Image verifications ( using mini-landing)
*Description of what is to be accomplished.*
I order to verify promo image overriding lead image, we need to add a tested content into a feature card on mini-landing page
1. Create a mini landing page
2. Add two item feature card row 
3. select newly cretaed content as a featured item 
4. save and publish 
5. verify promo is displayed in place of lead image 
**Parent Issue:** #9999

ADD BELOW SCENARIO TO $ACreationOfTestResources.feature
```gherkin
Scenario: User is creating new Mini Landing page content type
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on "Add content" action button
        And user clicks on "Mini Landing Page" content type
        Then page title is "Mini Landing Page"
        When user selects test site section
        And user fills out the following fields
            | fieldLabel       | value                                             | field_name             |
            | Pretty URL       | mini-landing-page-test-promo                      | field_pretty_url       |
            | Page Title       | Test Resource Mini Landing Page                  | title                  |
            | Browser Title    | Test Resource Mini Landing Page - Browser Title  | field_browser_title    |
            | Meta Description | Test Resource Mini Landing Page Meta Description | field_page_description |
           | Feature Card Description | Test Resource Mini Landing Page - Feature Card Desc | field_feature_card_description |       
  | Card Title               | Test Resource Mini Landing Page - Card Title        | field_card_title               | 
And user enters "Mini Landing Page List Description" into "List Description" text field
And user selects 1 Promotional Image for the mini landing 
 And user remembers the source of selected promo image for the mini landing
        And user selects "Published" from "Save as" dropdown
        Then user saves the content page
```
ADD BELOW SCENARIOS TO EVERY EXISTING CONTENT TESTS THAT HAVE A PROMO IMAGE SELECTED

```gherkin
    Scenario: Add a featured item to mini landing page
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url "mini-landing-page-test-promo" from the list of content
        And user clicks on the tool bar status green button "Published"
        And user clicks "View in edit form" button from other actions
        And user selects "Add Two Item Feature Card Row" from "Contents" dropdown
        And user clicks on "Featured Item" link in the "Internal Feature Card" text area
        And user clicks on "Select content" button in the "Internal Feature Card" text area
        And user selects "**replace this with a newly created content title**" item from the list
        Then user saves the content page
        And user clicks on the tool bar status green button "Editing"
        And user selects "Quick Publish" from workflow actions

    Scenario: Verify promo and card titles in mini landing
        Given user is navigating to the front end site with path site section plus "mini-landing-page-test-promo"
        Then page title is "Test Resource Mini Landing Page"
        Then the promo image is matching the earlier selected image


    Scenario: remove featured item
        Given user is navigating to "/user/login"
        When user enters credentials
        And user clicks "Log in" button
        Then user is logged in and the user name "admin" is displayed in the toolbar
        And the tool bar appears at the top
        When user clicks on "Content" tab
        And user clicks on title with url "mini-landing-page-test-promo" from the list of content
        And user clicks on the tool bar status green button "Published"
        And user clicks "View in edit form" button from other actions
        And user removes "Two Item Feature Card Row" section
        And user confirms removal
        Then user saves the content page
        And user clicks on the tool bar status green button "Editing"
        And user selects "Quick Publish" from workflow actions
```

ADD THE FOLLOWING STEPS at the end of scenario in $ZCleanUpTestContent.feature

```gherkin
 When user clicks on "Content" tab
              And user selects a checkbox next to title with url "mini-landing-page-test-promo" from the list of content
              And user clicks on "Apply to selected items" content action button
              Then page title is "Are you sure you want to delete this content item?"
              When user clicks on "Delete" button
              Then the confirmation text "Deleted 1 content item." appears on a screen
              And the content item with url "mini-landing-page-test-promo" does not exist in the list of content
```
### Sub-Tasks
- [ ] Create Component
- [ ] Implement Unit Tests for Component

### Notes

*Some complementary notes if necessary:*
* > Here goes a quote from an email
* Here goes whatever useful information can exist…
* More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by alinai**
Closes #248 

Updated the following files according to #248 : 
$ACreationOfTestResources.feature, Article.feature, $ZCleanUpTestContent.feature

### CODE REVIEW

**Comment by alinai** [comment]
Cypress test run on dev-acsf:

image
image
image
image
image
image
image
image

**Comment by alinai** [comment]
npm test run on dev-ac:
image

**Comment by alinai** [comment]
@KateMashkinaNIH ,please review.

File: cypress/e2e/CMS/$Article/index.js [hunk 1]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
+And('user clicks on {string} button item', (title) => {
+    cy.get(`input[value="${title}"]`).click({ force: true })
+});
 
+And('user selects {string} item from main page content', (title) => {
+    cy.getIframeBody('iframe.entity-browser-modal-iframe').find(`td:contains("${title}")`).parent().find('input').check();
+});
 
+Then('the promo image is matching the earlier selected image', () => {
+    function extractImgName(imageFullName) {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:191]: helper function should never be a part of a step - you should put it in utils folder and import where needed

File: cypress/e2e/CMS/$Article/index.js [hunk 2]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
+    const extractedImageName = extractImgName(expectedSrc).replace(/\.jpg|\.jpeg|\.png/, '')
 
+    cy.get('div.feature-card').find('img').then($el => {
+        const source = $el[0].getAttribute('src');
+        const actSrc = source.replace(/\?itok=[\S]+/, '').replace(/^(.*?)\/public/, '')
+        expect(actSrc).to.include(extractedImageName.replaceAll('_', '-').replace('article', ''))
+    })
+});
 
+And('user selects the {string} content item', (dropDown) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:217]: this step is duplicated 8 times - the code and locators are the same, no reason to copy paste it so many times, put it in the commons

File: cypress/e2e/CMS/$Article/index.js [hunk 3]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
+    cy.get('li.dropbutton-toggle').click();
+    cy.get(`input[id*="edit-field-landing-contents-add-more-add-more-button-cgov-two-item-feature-row"]`).click();
+});
 
+And('user clicks on the {string} link in the {string} text area', (title, cardOption) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:222]: same here - move it to common

File: cypress/e2e/CMS/$Article/index.js [hunk 4]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
+    cy.get(`summary.seven-details__summary span:contains('${title}')`).parent().click();
+});
 
+And('user removes {string} section', (removeSection) => {
+    cy.get('#edit-field-landing-contents-0-top-links-remove-button').click({ force: true });
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:227]: same here - move it to common

File: cypress/e2e/CMS/$Article/index.js [hunk 5]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
-
-
-
-
-
-
-
-
+And('user confirms removal', () => {
+    cy.get(`li.confirm-remove.dropbutton-action input[value='Confirm removal']`).click({ force: true });
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:231]: same here - move it to common

File: cypress/e2e/CMS/$Article/index.js [hunk 6]
```diff
@@ -27,6 +27,10 @@ And('user clicks on title with url {string} from the list of content', (contentH
     cy.get(`a[href='${siteSection}/${contentHref}-${randomNum}']`).click();
 });
 
+And('user clicks on the title with url {string} from the list of content', (contentHref) => {
+    cy.get(`a[href='${siteSection}/${contentHref}']`).click();
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:31]: same here - move it to commons

File: cypress/e2e/CMS/$Article/index.js [hunk 7]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
 
+And('user selects {string} content item', (dropDown) => {
+    cy.get(`input[value='${dropDown}']`).click({ force: true });
+});
 
+And('user clicks on {string} link in the {string} text area', (title, cartOption) => {
+    cy.get(`div.paragraph-type-title:contains('${cartOption}')`).parent().parent().find(`span:contains('${title}')`).parent().click();
+});
 
+And('user clicks on {string} button item', (title) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:182]: same here - move to common

File: cypress/e2e/CMS/$Article/index.js [hunk 8]
```diff
@@ -164,28 +171,62 @@ And('public use text is not displayed', () => {
...
+    cy.get(`input[value="${title}"]`).click({ force: true })
+});
 
+And('user selects {string} item from main page content', (title) => {
```
Comment (KateMashkinaNIH) [inline cypress/e2e/CMS/$Article/index.js:186]: same here - move to common

**Review by KateMashkinaNIH — CHANGES_REQUESTED** [review]

File: cypress/e2e/CMS/$Article/index.js (hunk 8 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:186]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 7 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:182]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 6 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:31]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 5 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:231]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 4 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:227]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 3 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:222]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 2 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js:217]: done

File: cypress/e2e/CMS/$Article/index.js (hunk 1 above)
Comment (alinai) [reply cypress/e2e/CMS/$Article/index.js]: done

**Comment by alinai** [comment]
cypress test run on dev-ac after updating per review:
image
image
image
image
image

**Comment by alinai** [comment]
Cypress test run on dev-acsf:
image
image
image
image
image
image
image
image
image

**Comment by alinai** [comment]
@KateMashkinaNIH , please review again. Thanks!

**Comment by alinai** [comment]
Cypress test run on dev-ac:
image

@KateMashkinaNIH , please review.

**Review by KateMashkinaNIH — APPROVED** [review]

**Pull request merged**